From fc1508b3ed47f8b7a5338196feb6ce14ad9491e5 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Wed, 21 May 2025 10:18:00 +0300
Subject: [PATCH] Revert "[SW-224361]: add retry in case spi_transceive fails"

This reverts commit 29cfb62131ff6036d3c12b7c5fc938f52a2e7cf8.

This commit is being reverted as SPI transceive failures are still
observed intermittently despite the changes.

A different fix, tracked using [SW-229103], is being applied to address
the issue more effectively.

Change-Id: I9e5ffdf417e1d6d02ec7d209e7d606e4597e5890
---

diff --git a/drivers/tpm/tpm2_tis_spi.c b/drivers/tpm/tpm2_tis_spi.c
index b7adc73..70312b3 100644
--- a/drivers/tpm/tpm2_tis_spi.c
+++ b/drivers/tpm/tpm2_tis_spi.c
@@ -48,7 +48,6 @@
 #define TPM_RID(l)			(0x0F04 | ((l) << 12))
 
 #define MAX_SPI_FRAMESIZE	32
-#define SPI_XFER_RETRY		5
 
 /* Number of wait states to wait for */
 #define TPM_WAIT_STATES		100
@@ -94,7 +93,6 @@
 	int transfer_len, ret, i;
 	uint8_t tx_bytes[MAX_SPI_FRAMESIZE];
 	uint8_t rx_bytes[MAX_SPI_FRAMESIZE];
-	uint8_t try = 0;
 	struct spi_buf tx_buf = {
 		.buf = tx_bytes,
 	};
@@ -170,14 +168,8 @@
 			memset(tx_bytes, 0, transfer_len);
 		}
 
-		while (try++ < SPI_XFER_RETRY) {
-			ret = spi_transceive(data->spi_dev, &config->spi_cfg,
-						&tx, in ? &rx : NULL);
-			if (!ret)
-				break;
-			LOG_INF("spi_transceive failed (%d)...retrying %d", ret, try);
-		}
-
+		ret = spi_transceive(data->spi_dev, &config->spi_cfg,
+				     &tx, in ? &rx : NULL);
 		if (ret) {
 			LOG_ERR("%s: spi read transfer failed (err: %d)\n",
 				__func__, ret);
